applications/ConvectionDiffusionApplication/python_scripts/convection_diffusion_solver.py

```diff
@@ -2,13 +2,25 @@
 
 # Importing the Kratos Library
 import KratosMultiphysics
-import KratosMultiphysics.python_linear_solver_factory as linear_solver_factory
-import KratosMultiphysics.base_convergence_criteria_factory as convergence_criteria_factory
 
 # Import applications
 import KratosMultiphysics.ConvectionDiffusionApplication
 import KratosMultiphysics.ConvectionDiffusionApplication.check_and_prepare_model_process_convection_diffusion as check_and_prepare_model_process
 
+# If required, import parallel applications and modules
+if KratosMultiphysics.ParallelEnvironment.GetDefaultDataCommunicator().IsDistributed():
```

Comment:
with the changes to how the data-comm should be used this is quite an unfortunate design

I am not blocking but IMO this should be done differently

Reply:
Agree, but as I said, seems that we are stuck with this for half a year already.
I don't remember if the changes for the GetDataCommunicator("Default") where merged already (or even implemented). If they were you are right, this should be change to explicitly request for the default comm at user level.

Anyway, seems that we were steering towards encapsulating mpi details outside thee user sight, so probably better to do that change when we make the final design.